### ISSUE TITLE
[VLCKit/Swiftfin Player] Audio stream index (?) incorrect during transcode

### ISSUE THREAD

**Issue by holow29** [jellyfin/Swiftfin#926]
### Describe the bug

During transcoding/DirectStreaming using VLCKit/Swiftfin player, selected audio track index will be incorrect, causing no audio to play by default. To get audio to play, it appears that the next track in the index needs to be selected. The server is only sending the one track to begin with.

For example, say a file has two audio tracks: x, y.
The track selection will list: none, x, y.
Currently, 'none' will be selected by default graphically while selected track index sent to the server will indicate 'x'. Server will transcode 'x' and send that back to client. Client appears to have something other than 'x' selected, though, causing no audio to play. To get audio, 'y' needs to be selected graphically and this will actually play the transcoded 'x'.

A workaround for now is here: https://github.com/sy6sy2/Swiftfin/commit/029af180781bfe63bc7b172437257cfd5e2bccfb VideoPlayerViewModel.swift changes seem to allow audio to play by default, but I am not sure the implications for DirectPlay.

Might be related to #771 

### Application version

main eb17300

### Where did you install the app from?

None

### Device information

Multiple simulators and devices

### OS version

iOS 16/17, tvOS 16/17

### Jellyfin server version

380c367

**Comment by LePips** [comment]
Yep, this is the issue that's causing the "no audio/wrong audio track", however I've also seen it on instances of direct play.

**Comment by holow29** [comment]
Doing some basic investigation on this: when VLCUI is building the audioTrackIndex in VLCMediaPlayerExtensions.swift Line 84, 'indexes' array seems to be {[index: -1,...],[index: 2,...]}. The question is why this is happening and how to solve for it. The expected behavior in a certain sense would be for audioTrackIndexes to look like: {[index: -1,...],[index: 1,...]},[index: 2,...],etc.}
-1 is the disabled state for audio, which is what we are getting by default now because we want track with index 1, but it incorrectly appears with index 2 in the array. Therefore, when it searches for track with index 1 (passed with AudioStreamIndex during VLC configuration or with track change button), nothing is found and it returns index -1 so no audio.

I need to do more investigation during DirectPlay to see how that is handled as well.

**Comment by 84480138** [comment]
大佬我有的视频要外网播放要转很久，这个应该也是音频解码的问题吗？可否帮忙看看，我的联系qq:84480138，麻烦大佬了

**Comment by swiftbird07** [comment]
Any progress here?
